backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/transformer/CollectiveAgreementTransformer.java

```diff
@@ -0,0 +1,44 @@
+package de.bund.digitalservice.ris.caselaw.adapter.transformer;
+
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.CollectiveAgreementDTO;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.CollectiveAgreementIndustryDTO;
+import de.bund.digitalservice.ris.caselaw.domain.CollectiveAgreement;
+
+public class CollectiveAgreementTransformer {
+  private CollectiveAgreementTransformer() {}
+
+  public static CollectiveAgreement transformToDomain(
+      CollectiveAgreementDTO collectiveAgreementDTO) {
+    if (collectiveAgreementDTO == null) {
+      return null;
+    }
+
+    return CollectiveAgreement.builder()
+        .id(collectiveAgreementDTO.getId())
+        .name(collectiveAgreementDTO.getName())
+        .date(collectiveAgreementDTO.getDate())
+        .norm(collectiveAgreementDTO.getNorm())
+        .industry(
+            CollectiveAgreementIndustryTransformer.transformToDomain(
+                collectiveAgreementDTO.getIndustry()))
+        .build();
+  }
+
+  public static CollectiveAgreementDTO transformToDTO(CollectiveAgreement collectiveAgreement) {
+    if (collectiveAgreement == null) return null;
+
+    var builder =
+        CollectiveAgreementDTO.builder()
+            .id(collectiveAgreement.id())
```

Comment:
Vielleicht ist das schon die Lösung für den Bug oben...
```suggestion
            .id(collectiveAggreement.newEntry() ? null : collectiveAggreement.id())
```